vision/detect_faces.rb

```diff
@@ -0,0 +1,80 @@
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in write, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START all]
+# [START import_client_library]
+require "google/cloud/vision"
+# [END import_client_library]
+# [START import_rmagick]
+require "rmagick"
+# [END import_rmagick]
+
+# [START detect_faces]
+def detect_faces path_to_image_file:, path_to_output_file:
+  # [START get_vision_service]
+  vision = Google::Cloud::Vision.new
+  # [END get_vision_service]
+
+  # [START detect_face]
+  image      = vision.image path_to_image_file
+  annotation = vision.annotate image, faces: true
```

Comment:
Not required ... but do you want to remove all `annotation`s from the code, while we're updating it?

I previously didn't see that [image](annotation = vision.annotate image, labels: true) has [`#faces`](https://googlecloudplatform.github.io/google-cloud-ruby/#/docs/google-cloud/v0.23.0/google/cloud/vision/image?method=faces-instance), [`#labels`](https://googlecloudplatform.github.io/google-cloud-ruby/#/docs/google-cloud/v0.23.0/google/cloud/vision/image?method=labels-instance), and [`#landmark`](https://googlecloudplatform.github.io/google-cloud-ruby/#/docs/google-cloud/v0.23.0/google/cloud/vision/image?method=landmark-instance)

Code like `image.faces` is *soooo* much more obvious than this unexpected `annotation` object